src/HexagonOptimize.cpp

```diff
@@ -214,12 +220,17 @@ Expr apply_patterns(Expr x, const std::vector<Pattern> &patterns, IRMutator *op_
                 if (p.flags & Pattern::SwapOperands) {
```

Comment:
I think the SwapOperands handler should go last (after the LosslessCastOp handler). I was thinking of SwapOperands as swapping prior to substitution, so it wouldn't affect the operand index of LosslessCastOp